File: api/Fitr/Models/Exercise.cs
using System.Collections.Generic;

namespace Fitr.Models
{
    public class Exercise
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public virtual ICollection<Set> Sets { get; set; }
        
        public virtual ICollection<WorkoutHasExercise> Workouts { get; set; }
    }
}

File: lib/models/exercise.dart
import 'package:fitr/models/exercise_set.dart';

class Exercise {
  int id;
  String name;
  List<ExerciseSet> sets;

  Exercise({this.id, this.name, this.sets});

  factory Exercise.fromJson(Map<String, dynamic> json) {
    var listOfSets = json['sets'] as List;
    var sets = listOfSets.map((set) => ExerciseSet.fromJson(set)).toList();

    return Exercise(id: json['id'], name: json['name'], sets: sets);
  }
}
